include/boost/json/basic_parser_impl.hpp

```diff
@@ -144,14 +98,23 @@ dec_to_float(
     std::int32_t e,
     bool neg) noexcept
 {
-    auto x =
-        static_cast<double>(m);
+    auto x = static_cast<double>(m);
     if(neg)
         x = -x;
+
     if(e < -305)
-        return x * 1e-305 *
-            pow10(e+305);
-    return x * pow10(e);
+    {
+        e += 305;
+        x *= 1e-305;
+    }
+
+    if(e < -308)
+        return x * 0.0;
```

Comment:
I don't know if division by infinity can trap on some platforms/configurations. If it doesn't, this check can be removed.

Reply:
I don't understand this line, why not just `return 0` ?

Reply:
Are negative zeros important?

Reply:
Actually, I think,  the whole block can be removed. This will result in `x / INFINITY`.